### PR TITLE
Issue #719 Performance regression with fragment additons to classpath #2

### PR DESCRIPTION
**Pull request by sratz**
Based on #721 with additional optimization.

### CODE REVIEW

**Comment by laeubi** [comment]
@sratz thanks for this improvement, I also noticed that there is an "resolved IU" that has attached fragments so we probably can use that in the future, but I think for now with these enhancements we should be fine as a first step :+1:

**Comment by sratz** [comment]
Branch should be up-to-date.

I'm trying to make sense of the test failures and if/why they are caused by this change.

File: tycho-bundles/org.eclipse.tycho.p2.resolver.impl/src/main/java/org/eclipse/tycho/p2/resolver/P2ResolverFactoryImpl.java [hunk 1]
```diff
@@ -134,33 +132,34 @@ public P2ResolverImpl createResolver(MavenLogger logger) {
...
-            Set<IInstallableUnit> dependencyFragments, IInstallableUnit unitToMatch) {
-        Iterator<Entry<IInstallableUnit, IRequiredCapability>> iterator = fragmentsList.iterator();
-        while (iterator.hasNext()) {
-            Entry<IInstallableUnit, IRequiredCapability> fragment = iterator.next();
-            if (fragment.getValue().isMatch(unitToMatch)) {
-                dependencyFragments.add(fragment.getKey());
-                iterator.remove();
-            }
-        }
+        Map<String, List<IInstallableUnit>> availableIUsById = availableIUs.stream()//
```
Comment (laeubi) [inline tycho-bundles/org.eclipse.tycho.p2.resolver.impl/src/main/java/org/eclipse/tycho/p2/resolver/P2ResolverFactoryImpl.java:136]:
```suggestion
        Map<String, List<IInstallableUnit>> availableIUsById = resolvedUnits.stream()//
```

Comment (sratz) [reply tycho-bundles/org.eclipse.tycho.p2.resolver.impl/src/main/java/org/eclipse/tycho/p2/resolver/P2ResolverFactoryImpl.java:136]: Oops :)

**Pull request merged**